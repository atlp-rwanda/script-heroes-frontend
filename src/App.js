import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Container } from "reactstrap";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fab } from "@fortawesome/free-brands-svg-icons";
import { faTrashAlt, faEdit } from "@fortawesome/free-solid-svg-icons";
import Home from "./components/HomePage";
import Login from "./views/login";
import Signup from "./Views/Signup";
import ViewProfile from "./components/profile/View";
import CompleteProfile from "./components/profile/Complete";
import UpdateProfile from "./components/profile/Update";
import Navigation from "./components/Navigation";
import Footer from "./components/Footer";
import Roles from "./views/Roles/Roles";

import "./App.scss";
import "jquery/dist/jquery.min.js";
import "bootstrap/dist/js/bootstrap.min.js";
import "bootstrap/dist/css/bootstrap.min.css";

library.add(fab, faTrashAlt, faEdit);
import AdminRequests from './components/requests/adminRequests/AdminRequests'

const App = () => (
  <div className="app">
    <Router>
      <Navigation />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/signup" component={Signup} />
          <Route exact path="/profile" component={ViewProfile} />
          <Route
            exact
            path="/profile/complete/:slug"
            component={CompleteProfile}
          />
          <Route exact path="/request/manager" component={AdminRequests} />
          <Route exact path="/profile/update/:slug" component={UpdateProfile} />
          <Route exact path="/roles" component={Roles} />
        </Switch>
      <Footer />
    </Router>
  </div>
);

export default App;
