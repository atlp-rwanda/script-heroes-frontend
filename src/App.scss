* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  .app {
    display: flex;
    flex-direction: column;
    height: 100vh;
    align-items: center;
    justify-content: center;
    background-color: #728191;
    color: #f9a826;
  }
  .title {
    font-size: 40px;
  }
     