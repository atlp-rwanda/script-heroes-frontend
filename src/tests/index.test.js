import React from "react";
import renderer from "react-test-renderer";

import App from "../App";

jest.mock("../components/profile/Complete", () => "<CompleteProfile/>");
jest.mock("../components/profile/Update", () => "<UpdateProfile/>");

describe("App", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
