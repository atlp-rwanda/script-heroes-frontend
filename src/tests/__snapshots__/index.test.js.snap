// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="app"
>
  <div
    className="container-fluid"
  >
    <div
      className="navigation-title row"
    >
      <div
        className="col"
      >
        <h1
          className="nav-title text-warning"
        >
          barefoot NOMAD
        </h1>
      </div>
      <div
        className="col"
      >
        <ul
          className="center-ele"
        >
          <form>
            <select
              className="lang-form"
            >
              <option
                value="english"
              >
                English
              </option>
              <option
                value="kinyarwanda"
              >
                Kinyarwanda
              </option>
              <option
                value="français"
              >
                Français
              </option>
            </select>
          </form>
        </ul>
      </div>
    </div>
    <nav
      className="navbar navbar-expand-lg navbar-light custom-nav"
    >
      <a
        className="navbar-brand mr-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo"
          src=""
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav ml-5"
        >
          <li
            className="nav-item active"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/"
              onClick={[Function]}
            >
              Home
              <span
                className="sr-only"
              >
                (current)
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/"
              onClick={[Function]}
            >
              Travel Tips
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/login"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li
            className="nav-item text-warning"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/"
              onClick={[Function]}
            >
              Services
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/"
              onClick={[Function]}
            >
              Contact
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-warning ml-4"
              href="/"
              onClick={[Function]}
            >
              <button
                className="book-btn"
              >
                Book now
              </button>
            </a>
          </li>
          <input
            className="search-input"
            placeholder=""
            type="search"
          />
        </ul>
      </div>
    </nav>
  </div>
  <div
    className="container"
  >
    <div
      className="home-flex d-flex row"
    >
      <div
        className="home-intro-text col"
      >
        <h1>
          Let us take you to your happy place
        </h1>
        <p>
          Welcome! At Barefoot Nomad, we value customers. We believe that customers' happiness is our happiness. Do not hesitate to work with us. Hit the button below to get started.
        </p>
        <a
          className="start-btn"
          href="/login"
          onClick={[Function]}
        >
          Get Started
        </a>
      </div>
      <div
        className="col"
      >
        <img
          alt="Destination"
          className="destination"
          src=""
        />
      </div>
    </div>
    <h1
      className="pop-services"
    >
      Our Popular Services
    </h1>
    <div
      className="services"
    >
      <button
        className="service-btn prev-btn"
        data-testid="prev"
        disabled={false}
        onClick={[Function]}
      >
        <img
          src=""
        />
      </button>
      <div
        className="page"
      >
        <div
          className="custom-col"
        >
          <div
            className="cards-slider active-slide-3"
          >
            <div
              className="cards-slider-wrapper"
              style={
                Object {
                  "transform": "translateX(-30%)",
                }
              }
            >
              <div
                className="card"
                id="card-0"
              >
                <img
                  alt="Hotel booking"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Hotel booking
                  </h1>
                  <p
                    className="description"
                  >
                     At Barefoot Nomad, you can book any hotel you want from an incredible list of hotels available worldwide at incomparable prices.
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-1"
              >
                <img
                  alt="Vacation booking"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Vacation booking
                  </h1>
                  <p
                    className="description"
                  >
                    Ever worried on how your vacation will be spent? Worry no more! Barefoot Nomad is here to make your vacation memorable than ever.
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-2"
              >
                <img
                  alt="Flight booking"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Flight booking
                  </h1>
                  <p
                    className="description"
                  >
                    Barefoot Nomad offers booking services in over 150 destinations worldwide. Book yours today and benefit to work with a modernized company
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-3"
              >
                <img
                  alt="Travel Advisory"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Travel Advisory
                  </h1>
                  <p
                    className="description"
                  >
                    In dilemma failing to decide where to go first? Barefoot Nomad will advise you on how you can organize your travels and make everything work as expected
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-4"
              >
                <img
                  alt="Travel Consultancy"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Travel Consultancy
                  </h1>
                  <p
                    className="description"
                  >
                    We're here for you. Invite Barefoot Nomad in your organization so we can help you in planning, excecusion, and following up your travels and vacations.
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-5"
              >
                <img
                  alt="Hotel booking"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Hotel booking
                  </h1>
                  <p
                    className="description"
                  >
                     At Barefoot Nomad, you can book any hotel you want from an incredible list of hotels available worldwide at incomparable prices.
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-6"
              >
                <img
                  alt="Vacation booking"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Vacation booking
                  </h1>
                  <p
                    className="description"
                  >
                    Ever worried on how your vacation will be spent? Worry no more! Barefoot Nomad is here to make your vacation memorable than ever.
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-7"
              >
                <img
                  alt="Flight booking"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Flight booking
                  </h1>
                  <p
                    className="description"
                  >
                    Barefoot Nomad offers booking services in over 150 destinations worldwide. Book yours today and benefit to work with a modernized company
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-8"
              >
                <img
                  alt="Travel Advisory"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Travel Advisory
                  </h1>
                  <p
                    className="description"
                  >
                    In dilemma failing to decide where to go first? Barefoot Nomad will advise you on how you can organize your travels and make everything work as expected
                  </p>
                </div>
              </div>
              <div
                className="card"
                id="card-9"
              >
                <img
                  alt="Travel Consultancy"
                  className="service-img"
                  src=""
                />
                <div
                  className="details"
                >
                  <h1
                    className="card-title"
                  >
                    Travel Consultancy
                  </h1>
                  <p
                    className="description"
                  >
                    We're here for you. Invite Barefoot Nomad in your organization so we can help you in planning, excecusion, and following up your travels and vacations.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="service-btn next-btn"
        data-testid="next"
        disabled={false}
        onClick={[Function]}
      >
        <img
          src=""
        />
      </button>
    </div>
  </div>
  <div
    className="footer container-fluid"
  >
    <div
      className="just-foot row"
    >
      <div
        className="social-links row"
      >
        <div
          className="col"
        >
          <a
            href="https://www.facebook.com/"
            target="_blank"
          >
            <img
              src=""
            />
          </a>
          <a
            href="http://instagram.com/"
            target="_blank"
          >
            <img
              src=""
            />
          </a>
          <a
            href="http://linkedin.com/"
            target="_blank"
          >
            <img
              src=""
            />
          </a>
          <a
            href="https://twitter.com/"
            target="_blank"
          >
            <img
              src=""
            />
          </a>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <p>
            © 2020 Barefoot Nomad
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
